graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/KotlinDataTypeGenerator.kt

```diff
@@ -97,7 +97,7 @@ class KotlinInputTypeGenerator(config: CodeGenConfig, document: Document) :
                 }
             )
         val interfaces = emptyList<Type<*>>()
-        return generate(definition.name, fields, interfaces, config.implementSerializable, document, definition.description)
+        return generate(definition.name, fields, interfaces, document, definition.description)
     }
 
     private fun generateCode(value: Value<Value<*>>, type: KtTypeName): CodeBlock =
```

Comment:
This method consolidates the main changes that this PR addresses.
* Read the `implementSerializable` and `generateAllConstructor` from the `config` reference already available.
* Modify how we declare _type properties_ and _type constructor parameters_ such that the _type properties_ are declared as part of the default constructor, e.g. `Car(val foo: String = null)`.
* If `generateAllConstructor` is disabled, we are making the _Kotlin Data Type_ mutable such that it respects the semantics of _creating a new reference with the empty constructor and then set the properties_.